### PR TITLE
template_experiment_config with __init__ and evalutate_genomes functions

### PR DESCRIPTION
**Pull request by aidanhor-n**

### CODE REVIEW

File: neat/experiments/template/config.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+        self.CONNECTION_MUTATION_RATE = kwargs['CONNECTION_MUTATION_RATE']
+        self.CONNECTION_PERTURBATION_RATE = kwargs['CONNECTION_PERTURBATION_RATE']
+        self.ADD_NODE_MUTATION_RATE = kwargs['ADD_NODE_MUTATION_RATE']
+        self.ADD_CONNECTION_MUTATION_RATE = kwargs['ADD_CONNECTION_MUTATION_RATE']
+
+        self.CROSSOVER_REENABLE_CONNECTION_GENE_RATE = kwargs['CROSSOVER_REENABLE_CONNECTION_GENE_RATE']
+
+        # Top percentage of species to be saved before mating
+        self.PERCENTAGE_TO_SAVE = kwargs['PERCENTAGE_TO_SAVE']
```
Comment (GregoryConrad) [inline neat/experiments/template/config.py:46]: assuming you are given a dict of all the parameters in kwargs, something like the following would be a great simplification of this:
```python3
for k, v in kwargs.items():
  setattr(self, k, v)
```
This is also not error-prone to a typo in a string or forgetting items.

File: neat/experiments/template/config.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    #         # Compute the loss
+    #         loss = nn.functional.mse_loss(pred, target)
+
+    #         # Compute the fitness
+    #         fitness -= loss.item()
+    #         # loss = criterion(pred, target)
+
+    #     return fitness
+
+    def eval_genomes(self, genomes):
```
Comment (GregoryConrad) [inline neat/experiments/template/config.py:86]: NOTE: I have not reviewed this particular method yet b/c of the lot of TODOs. Please clean up the other areas I mentioned and complete those TODOs and then I will take a look at this method

**Review by GregoryConrad — CHANGES_REQUESTED** [review]

File: images/pole-balancing-solution [hunk 3]
```diff
@@ -1,12 +1,8 @@
 digraph {
 	node [fontsize=9 height=0.2 shape=circle width=0.2]
```
[the remark this answers is not in the document]
Comment (GregoryConrad) [reply images/pole-balancing-solution:2]: This file is still not the same as the one on master; please revert

**Review by GregoryConrad — CHANGES_REQUESTED** [review]
Please undo the changes to /images/pole-balancing-solution and move the pole_run file back to it's original place

File: neat/experiments/template/default_kwarg [hunk 4]
```diff
@@ -0,0 +1,24 @@
+default_kwarg = {
```
[the remark this answers is not in the document]
Comment (GregoryConrad) [reply neat/experiments/template/default_kwarg:1]: Also, please change the file extension to .py

File: neat/experiments/template/config.py (hunk 1 above)
Comment (GregoryConrad) [reply neat/experiments/template/config.py:46]: Please change __init__ to use this.

File: neat/experiments/template/config.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+from neat.phenotype.feed_forward import FeedForwardNet
+
+from utils import create_prediction_map, random_ensemble_generator_for_static_genome
+
+
+class TemplateConfig:
+
+    #TODO define __repr__(self) to return experiment parameters for SA
+
+    def __init__(self, **kwargs):
```
[the remark this answers is not in the document]
Comment (GregoryConrad) [reply neat/experiments/template/config.py:13]: how is kwargs going to be used?

**Review by GregoryConrad — CHANGES_REQUESTED** [review]

File: neat/experiments/template/config.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (aidanhor-n) [reply neat/experiments/template/config.py:13]: yes, the plan is to deep copy DEFAULT_KWARGS and change it for experiment specific parameters.

**Review by GregoryConrad — APPROVED** [review]

**Pull request merged**